scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -26,6 +26,9 @@
 if not ZEPHYR_BASE:
     sys.exit("$ZEPHYR_BASE environment variable undefined")
 
+ZEPHYR_TOOLCHAIN_VARIANT = os.getenv("ZEPHYR_TOOLCHAIN_VARIANT",
```

Comment:
we have this information already provided to us by cmake, see get_toolchain in the same file. Can you get this from the environment by doing the discvery of toolchain before you parse the options?

Reply:
Done